src/Autocomplete/AutocompleteOverlay.tsx

```diff
@@ -58,14 +59,17 @@ function AutocompleteOverlay({
       ref={floatingElementRef as React.RefObject<HTMLDivElement>}
       top={position?.top}
       left={position?.left}
-      visibility={showMenu ? 'visible' : 'hidden'}
       sx={{
         overflow: 'auto',
       }}
       {...overlayProps}
     >
       {children}
     </Overlay>
+  ) : (
+    // HACK: This ensures AutocompleteMenu is still mounted when closing the menu and all of the hooks inside of it are still called.
+    // A better way to do this would be to move the hooks to AutocompleteOverlay or somewhere that won't get unmounted.
+    <VisuallyHidden>{children}</VisuallyHidden>
```

Comment:
Comparing the a11y tree both production and this branch, in production (left) the listbox is not in the a11y tree until the autocomplete is opened. While, in this branch, it's in the a11y tree even when the autocomplete is closed.

because Overlay uses the css property `visibility`, [it removes the contents from the accessibility tree](https://developer.mozilla.org/en-US/docs/Web/CSS/visibility#accessibility_concerns), which VisuallyHidden does not. 

<img width="1444" alt="in production (left) the listbox is not in the a11y tree until the autocomplete is opened. While, in this branch, it's in the a11y tree even when the autocomplete is closed" src="https://github.com/primer/react/assets/1863771/c0811d23-3229-4c2d-bde5-12c08fdc496d">


If we can rid of the hack, that would of course be perfect. But, if not, we probably would need to make it aria-hidden as well.